### DISCUSSION TITLE
Chapter 2 Data File - unable to open Zip file

### DISCUSSION THREAD

**Discussion by bluethunder808** [headfirstpython/third#1] [General]
Hi,

I downloaded swimdata.zip but when I attempt to open, my iMac says it is an unsupported format.  Can you verify that the file is okay?  Thanks!

**Comment by headfirstpython** [comment]
I have downloaded the ZIP file to my Mac laptop with no issues.  The file
is fine.  Thanks.

On Sat, 7 Oct 2023 at 21:32, bluethunder808 ***@***.***>
wrote:

> Hi,
>
> I downloaded swimdata.zip but when I attempt to open, my iMac says it is
> an unsupported format. Can you verify that the file is okay? Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/headfirstpython/third/discussions/1>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7O3GTCLRIQUUHTGUD3MFK3X6G37TAVCNFSM6AAAAAA5XE5372VHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZVG4YTCNJTGY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Paul Barry

BSc (Hons), MSc, PG Cert (T&L)

Lecturer - Department of Computing
E ***@***.*** | *setu.ie* <https://setu.ie> | work
E ***@***.*** | personal
[image: turn on images] <https://setu.ie>

Kilkenny Road Campus, Kilkenny Road, Carlow, R93 V960, Ireland
Campas Bhóthar Chill Chainnigh, Bóthar Chill Chainnigh, Ceatharlach, R93
V960, Éire

**Comment by bluethunder808** [comment]
I have no problems downloading it... it is when I attempt to unzip it.  Can you unzip the file?

**Comment by headfirstpython** [comment]
Yes. File unzips for me with no issues.

What happens when you download/unzip swimdata2.zip from chapter11 folder?

**Comment by bluethunder808** [comment]
I tried the swimdata2.zip file in the Chapter11 folder and got the same error message:

<img width="377" alt="Screen Shot 2023-10-08 at 8 47 08 PM" src="https://github.com/headfirstpython/third/assets/147279550/8a4c460f-e883-4c8f-a6d4-bd57a4296862">

But I explored other areas of the headfirstpython site on Github and came across a "view raw" link and now the swimdata.zip file downloads correctly now.  

<img width="1036" alt="Screen Shot 2023-10-08 at 8 48 47 PM" src="https://github.com/headfirstpython/third/assets/147279550/7d486bd9-a54c-4715-a0a1-93eb8b20afa5">

I was right-clicking the file and saving the link as a file and somehow it was getting corrupted.  Strange.

Thanks for the help, I think the view raw option is what I need to use from now on.

**Comment by headfirstpython** [comment]
OK. I believe you download the current webpage (or a snippet of HTML) when
you right-click and "save as".  But, yes, downloading the "raw" file is
what you want.  👍

On Mon, 9 Oct 2023 at 07:51, bluethunder808 ***@***.***>
wrote:

> I tried the swimdata2.zip file in the Chapter11 folder and got the same
> error message:
> [image: Screen Shot 2023-10-08 at 8 47 08 PM]
> <https://user-images.githubusercontent.com/147279550/273513341-8a4c460f-e883-4c8f-a6d4-bd57a4296862.png>
>
> But I explored other areas of the headfirstpython site on Github and came
> across a "view raw" link and now the swimdata.zip file downloads correctly
> now.
> [image: Screen Shot 2023-10-08 at 8 48 47 PM]
> <https://user-images.githubusercontent.com/147279550/273513547-7d486bd9-a54c-4715-a0a1-93eb8b20afa5.png>
>
> I was right-clicking the file and saving the link as a file and somehow it
> was getting corrupted. Strange.
>
> Thanks for the help, I think the view raw option is what I need to use
> from now on.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/headfirstpython/third/discussions/1#discussioncomment-7227199>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A7O3GTAZ44WVA2Z54RIR5TLX6ONI7AVCNFSM6AAAAAA5XE5372VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TEMRXGE4TS>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Paul Barry

BSc (Hons), MSc, PG Cert (T&L)

Lecturer - Department of Computing
E ***@***.*** | *setu.ie* <https://setu.ie> | work
E ***@***.*** | personal
[image: turn on images] <https://setu.ie>

Kilkenny Road Campus, Kilkenny Road, Carlow, R93 V960, Ireland
Campas Bhóthar Chill Chainnigh, Bóthar Chill Chainnigh, Ceatharlach, R93
V960, Éire

**Comment by shermansodev** [comment]
On Chapter 6, when run the app for the final code, got this error, but the templates folder is already created under the webapp folder. 
Error from Debug trace:
webapp\swimclub.py", line 82, in produce_bar_chart
with open(save_to, 'w') as tf:  # the tf alias refers to the opened file.
     ^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'templates/Abi-10-100m-Back.html'
